vars/piperStageWrapper.groovy

```diff
@@ -30,8 +30,7 @@ void call(Map parameters = [:], body) {
 
     stageLocking(config) {
         def containerMap = ContainerMap.instance.getMap().get(stageName) ?: [:]
-        if (Boolean.valueOf(env.ON_K8S) && containerMap.size() > 0) {
-            DebugReport.instance.environment.put("environment", "Kubernetes")
+        if (Boolean.valueOf(env.ON_K8S) && (containerMap.size() > 0 || config.runStageInPod)) {
```

Comment:
so `runStageInPod` is new, right? how would I use that? Is it not documented on purpose?

Reply:
Well as piperStageWrapper is currently more a helper step for the pipelines and this is more a pipeline internal feature, I did not document it.

Reply:
Seems like you are not using environment variable. Do we have to enable this in config file? How do we have to use this parameter?

Reply:
Idea would be to enable it anyway by default in the Cloud SDK Pipeline.

Reply:
Alternatively you could also inject it as you inject the Docker image for the Kubernetes agent.

Reply:
Maybe you could verify if this solve your issue. Just use the piper branch 
Use `library "piper-lib-os@always-spawn-pod"` instead or before of `loadPiper`.
And add
...
script { 
    commonPipelineEnvironment.configuration.general.runStageInPod = true
}
```
where you also initialize the jnlp agent.

Reply:
yes, it works 👍 but should explicitly set the above parameter to true.

Reply:
can you show me that code snippet after solution?

Reply:
> @srinikitha09 @daniel-kurzynski can you show me that code snippet after solution?

That parameter comes from cloud sdk pipeline and it is by default set to true.

Reply:
@srinikitha09, @daniel-kurzynski  fine. But what we are facing the issue is, after checkout the build stage is moving to Jenkins master , we want that to run in agent only. is the library file edit needed? can you please advise

Reply:
Are you using cloud sdk pipeline and which type of project is it?